aws_advanced_python_wrapper/limitless_connection_plugin.py

```diff
@@ -0,0 +1,531 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  Licensed under the Apache License, Version 2.0 (the "License").
+#  You may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from __future__ import annotations
+
+import copy
+import time
+from contextlib import closing
+from threading import Event, RLock, Thread
+from time import sleep
+from typing import (TYPE_CHECKING, Any, Callable, ClassVar, List, Optional,
+                    Set, Tuple)
+
+from aws_advanced_python_wrapper.database_dialect import (
+    AuroraLimitlessDialect, DatabaseDialect)
+from aws_advanced_python_wrapper.errors import (AwsWrapperError,
+                                                UnsupportedOperationError)
+from aws_advanced_python_wrapper.host_availability import HostAvailability
+from aws_advanced_python_wrapper.hostinfo import HostInfo, HostRole
+from aws_advanced_python_wrapper.plugin import Plugin
+from aws_advanced_python_wrapper.utils.concurrent import ConcurrentDict
+from aws_advanced_python_wrapper.utils.log import Logger
+from aws_advanced_python_wrapper.utils.messages import Messages
+from aws_advanced_python_wrapper.utils.properties import (Properties,
+                                                          WrapperProperties)
+from aws_advanced_python_wrapper.utils.sliding_expiration_cache import \
+    SlidingExpirationCacheWithCleanupThread
+from aws_advanced_python_wrapper.utils.telemetry.telemetry import (
+    TelemetryContext, TelemetryFactory, TelemetryTraceLevel)
+from aws_advanced_python_wrapper.utils.utils import LogUtils
+
+if TYPE_CHECKING:
+    from aws_advanced_python_wrapper.driver_dialect import DriverDialect
+    from aws_advanced_python_wrapper.pep249 import Connection
+    from aws_advanced_python_wrapper.plugin_service import PluginService
+
+logger = Logger(__name__)
+
+
+class LimitlessConnectionPlugin(Plugin):
```

Comment:
In Python we decided to drop the "Connection" substring because we thought it was unnecessary and it shortens the class name, can we do the following renames:
- limitless_connection_plugin.py -> limitless_plugin.py
- test_limitless_connection_plugin.py -> test_limitless_plugin.py
- LimitlessConnectionPlugin -> LimitlessPlugin
- LimitlessConnectionPluginFactory -> LimitlessPluginFactory
- changing the `LimitlessConnectionPlugin.*=[LimitlessConnectionPlugin] ...` messages to `LimitlessPlugin.*=[LimitlessPlugin] ...`

Reply:
I agree. Also changed LimitlessConnectionContext to LimitlessContext